torch/csrc/jit/tensorexpr/expr.h

```diff
@@ -165,13 +165,45 @@ class Var : public ExprNode<Var> {
   std::string name_hint_;
 };
 
+// An expression to construct the underlying variable node.
+// Note: do not store any info here, since it is often possible to slice this
+// object. For example: VarHandle x('x'); ExprHandle x2 = x;
+class VarHandle : public ExprHandle {
+ public:
+  VarHandle() : ExprHandle(nullptr) {}
+  explicit VarHandle(Dtype dtype) : ExprHandle(Var::make(dtype)) {}
+  VarHandle(const std::string& name_hint, Dtype dtype)
+      : ExprHandle(Var::make(name_hint, dtype)) {}
+  explicit VarHandle(const Var* node) : ExprHandle(node) {}
+  const Var* node() const {
+    return static_cast<const Var*>(ExprHandle::node());
+  }
+  bool operator==(const VarHandle& other) const {
+    return this->node() == other.node();
+  }
+  bool operator!=(const VarHandle& other) const {
+    return !(*this == other);
+  }
+
+  const std::string& name_hint() const {
+    return this->node()->name_hint();
+  }
+  bool empty() const {
+    return (this->node() == nullptr);
+  }
+};
+
 class TORCH_API Buf : public ExprNode<Buf> {
  public:
   static ExprHandle make(
       const std::string& name_hint,
       const std::vector<ExprHandle>& dims,
       Dtype dtype);
   static ExprHandle make(const std::vector<ExprHandle>& dims, Dtype dtype);
+  static ExprHandle make(
```

Comment:
I'm not sure this is necessary. We could create `Buf` objects directly without first creating `Var`s for them - these vars are usually not used anywhere.